### PR TITLE
feat: pass immutable info on files

### PR DESCRIPTION
**Pull request by bschlenk**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Passes an `immutable` flag through assetInfo which other plugins may use.

https://github.com/webpack-contrib/file-loader/issues/382

### CODE REVIEW

File: src/index.js
Comment (bschlenk) [inline src/index.js:18]: Ideally this would be another function in https://github.com/webpack/loader-utils

Comment (evilebottnawi) [reply src/index.js:18]: Yes, I will test it deeply in near future

**Comment by bschlenk** [comment]
not sure why [ModuleConcatenationPlugin test](https://github.com/webpack-contrib/file-loader/pull/383/checks?check_run_id=1031775282) is failing. [Source](https://github.com/webpack/webpack/blob/820174baef91f26839ee0cefa2cfb4c1785c3f68/lib/optimize/ModuleConcatenationPlugin.js) doesn't use the `immutable` flag, is it unrelated?

**Comment by evilebottnawi** [comment]
I think we should improve test, webpack@5 implements better tree so count of modules can be reduced

**Review by evilebottnawi — APPROVED** [review]
Thanks

**Comment by evilebottnawi** [comment]
I will merge and fix tests after merge

**Pull request merged**
